crates/runtime/src/instance/allocator/pooling/memory_pool.rs

```diff
@@ -297,7 +447,11 @@ impl MemoryPool {
             self.return_memory_image_slot(allocation_index, image);
         }
 
-        self.index_allocator.free(SlotId(allocation_index.0));
+        let stripe_index = allocation_index.index() % self.stripes.len();
+        let stripe_slot = allocation_index.0 / self.stripes.len() as u32;
```

Comment:
Since there's a method `as_unstriped_slot_index` could there perhaps be a sibiling method for converting in this direction instead of inlining here? Mostly in the hopes that keeping the two near each other would be helpful for future modifications